kitsune/search/v2/base.py

```diff
@@ -37,10 +41,10 @@ def prepare(cls, instance, parent_id=None, **kwargs):
         """
 
         obj = cls()
+
         doc_mapping = obj._doc_type.mapping
         fields = [f for f in doc_mapping]
         fields.remove("indexed_on")
-
         # Loop through the fields of each object and set the right values
         for f in fields:
```

Comment:
Can we put another `hasattr(instance, "es_discard_doc")` check here to conditionally skip the `for f in fields` loop? I can't think of a need to go over the fields preparing the document if we already know we want to delete it. The id is prepared outside the loop, and I think that's all we need.

Reply:
I thought of that but elasticsearch does a copy of the object and returns a 500 error because it cannot find the data it expects.

Reply:
When does that copy happen? I've wrapped the for loop in a `if not hasattr(instance, "es_discard_doc"):` block, and it works for me with `ProfileDocument`